### ISSUE TITLE
gist for unseen prompt/task

### ISSUE THREAD

**Issue by La-SilverLand** [jayelm/gisting#5]
hi,  the idea in your work is very impressive, congrats
i didn't find much detail on how to get the gist tokens for new tasks without training, can you elaborate on that a little bit ?

**Comment by jayelm** [comment]
Hi, in short, when you give a prompt to a model followed by a gist token, the model learns to compress information from the prompt into the transformer activations on top of the gist token. This compression is done via a forward pass through the transformer, and is not learned by modifying Transformer weights or training in the traditional sense.

That is, you can think of the Transformer (prior to the gist token) as a compression model which takes an arbitrary prompt $t$ and maps it into a compressed version, $G(t)$. After learning compression across a wide variety of training tasks $t \sim T$, the gist model will ideally be able to compress arbitrary unseen prompts with no additional training.

**Comment by La-SilverLand** [comment]
so, does the whole process work like this?
1. training stage, the gist tokens are trained by instruction tuning, the model weights and gist embeddings are updated, like the tradional training
2. inference stage, first the [prompt, gist tokens] is feed into the model to get gist tokens, and then the [gist tokens, query] is feed into the model to generate a response for the query.
so we need a task prompt for a new task, and a sequencial 2-step inference, right ?

**Comment by jayelm** [comment]
that's correct! obviously doing the 2 step inference is slower than doing the single forward pass; the speedup gains happen if you have a lengthy prompt that you expect to reuse many times with different inputs; then you can get the gist tokens only once, and don't need to redo the first compression step.
